[17.0][MIG] webservice: Migration to 17.0

john-herholz-dt opened the pull request.

etobella committed:
[ADD] webservice

oca-travis committed:
[UPD] Update webservice.pot

OCA-git-bot committed:
[UPD] README.rst

etobella committed:
[MIG] webservice: Migration to 14.0

oca-travis committed:
[UPD] Update webservice.pot

OCA-git-bot committed:
[UPD] README.rst

Yvesldff committed:
Added translation using Weblate (French)

Yvesldff committed:
Translated using Weblate (French)

Currently translated at 85.0% (17 of 20 strings)

Translation: edi-14.0/edi-14.0-webservice
Translate-URL: https://translation.odoo-community.org/projects/edi-14-0/edi-14-0-webservice/fr/

LoisRForgeFlow committed:
[FIX] webservice: server.env.mixin needs to be inherited

The funtionality is used by the module as the method
`_server_env_fields` is defined, but the inherit of the mixin
using it is missing.

oca-travis committed:
[UPD] Update webservice.pot

OCA-git-bot committed:
webservice 14.0.1.0.1

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-14.0/edi-14.0-webservice
Translate-URL: https://translation.odoo-community.org/projects/edi-14-0/edi-14-0-webservice/

JasminSForgeFlow committed:
[MIG] webservice: Migration to 15.0

Commit:
[UPD] Update webservice.pot

OCA-git-bot committed:
[UPD] README.rst

Commit:
[UPD] Update webservice.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-15.0/edi-15.0-webservice
Translate-URL: https://translation.odoo-community.org/projects/edi-15-0/edi-15-0-webservice/

simahawk committed:
webservice: move to web-api

OCA-git-bot committed:
[UPD] README.rst

simahawk committed:
webservice: improve call

'call' now accepts these explicit kw params:

* 'url': to override the request's url completely
* 'url_params': to pass interpolation params for url formatting.

simahawk committed:
webservice: add api key and public auth support

EvaSForgeFlow committed:
[MIG] webservice: Migration to 16.0

Commit:
[UPD] Update webservice.pot

OCA-git-bot committed:
[UPD] README.rst

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: web-api-16.0/web-api-16.0-webservice
Translate-URL: https://translation.odoo-community.org/projects/web-api-16-0/web-api-16-0-webservice/

OCA-git-bot committed:
[UPD] README.rst

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (43 of 43 strings)

Translation: web-api-16.0/web-api-16.0-webservice
Translate-URL: https://translation.odoo-community.org/projects/web-api-16-0/web-api-16-0-webservice/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (43 of 43 strings)

Translation: web-api-16.0/web-api-16.0-webservice
Translate-URL: https://translation.odoo-community.org/projects/web-api-16-0/web-api-16-0-webservice/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (43 of 43 strings)

Translation: web-api-16.0/web-api-16.0-webservice
Translate-URL: https://translation.odoo-community.org/projects/web-api-16-0/web-api-16-0-webservice/it/

JordiMForgeFlow committed:
[IMP] webservice: multi-company

Commit:
[UPD] Update webservice.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: web-api-16.0/web-api-16.0-webservice
Translate-URL: https://translation.odoo-community.org/projects/web-api-16-0/web-api-16-0-webservice/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (44 of 44 strings)

Translation: web-api-16.0/web-api-16.0-webservice
Translate-URL: https://translation.odoo-community.org/projects/web-api-16-0/web-api-16-0-webservice/it/

john-herholz-dt commented:
Having some difficulties right now with the tests.

`odoo.tests.common.BlockedRequest: External requests verboten (was POST http://demo.localhost.odoo/)`

Any help is appreciated

**EDIT**

I was able to mock the _request_handler method to not get blocked. Tests are passing now.

gurneyalex committed:
[IMP] webservice: add support for oauth2

Allow using oauth2 with Backend Application Flow / Client Credentials
Grant.

gurneyalex committed:
[IMP] webservice: combine the url with collection's url

when a call is made with just a path, combine it with the collection's url

better combination of urls (Co-authored-by: Simone Orsi <simahawk@users.noreply.github.com>):

OCA-git-bot committed:
[BOT] post-merge updates

gurneyalex committed:
add support for oauth2 web application flow

gurneyalex committed:
fixup! add support for oauth2 web application flow

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (44 of 44 strings)

Translation: web-api-16.0/web-api-16.0-webservice
Translate-URL: https://translation.odoo-community.org/projects/web-api-16-0/web-api-16-0-webservice/it/

Commit:
[UPD] Update webservice.pot

gurneyalex commented:
Can you include https://github.com/OCA/web-api/pull/35 and https://github.com/OCA/web-api/pull/47 in there?

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: web-api-16.0/web-api-16.0-webservice
Translate-URL: https://translation.odoo-community.org/projects/web-api-16-0/web-api-16-0-webservice/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (76 of 76 strings)

Translation: web-api-16.0/web-api-16.0-webservice
Translate-URL: https://translation.odoo-community.org/projects/web-api-16-0/web-api-16-0-webservice/it/

gurneyalex committed:
[FIX] webservice: WARNING message in logs

The use of a compute method on `oauth2_flow` when this field is touched by
the server environment mixin causes it to be defined twice as computed,
with differents settings, and this ultimately causes a warning message
in the logs:

```
WARNING odoo odoo.modules.registry: webservice.backend: inconsistent 'compute_sudo' for computed fields: protocol, url, auth_type, username, password, api_key, api_key_header, oauth2_flow, oauth2_clientid, oauth2_client_secret, oauth2_token_url, oauth2_authorization_url, oauth2_audience, oauth2_scope, content_type
```

We fix this by using an old fashioned onchange declaration on the
`auth_type` field.

john-herholz-dt committed:
[IMP] webservice: pre-commit auto fixes

john-herholz-dt committed:
[MIG] webservice: Migration to 17.0

john-herholz-dt committed:
[DON'T MERGE] test-requirements.txt

john-herholz-dt committed:
fix test request handler mock

john-herholz-dt committed:
fix test request handler mock

john-herholz-dt committed:
fix test request handler mock

john-herholz-dt committed:
fix test request handler mock

john-herholz-dt committed:
fix test request handler mock

john-herholz-dt committed:
fix test request handler mock

john-herholz-dt committed:
fix test request handler mock

john-herholz-dt commented:
I tried my luck today, but I don't understand the failing checks, to be honest. My knowledge in the requests library is limited.

john-herholz-dt committed:
fix test request handler mock

john-herholz-dt committed:
fix test request handler mock

john-herholz-dt committed:
fix test request handler mock

john-herholz-dt committed:
fix test request handler mock

john-herholz-dt committed:
fix test request handler mock

john-herholz-dt committed:
fix test request handler mock

john-herholz-dt committed:
fix test request handler mock
